packages/typespec-python/src/emitter.ts

```diff
@@ -83,44 +77,63 @@ export function getModelsMode(context: SdkContext): "msrest" | "dpg" | "none" {
     return "dpg";
 }
 
-function addDefaultCalculatedOptions(
-    sdkContext: SdkContext,
-    options: PythonEmitterOptions & InternalPythonEmitterOptions,
-    yamlMap: Record<string, any>,
-) {
+function addDefaultOptions(sdkContext: SdkContext) {
+    const defaultOptions = {
+        "package-version": "1.0.0b1",
+        "generate-packaging-files": true,
+    };
+    sdkContext.emitContext.options = {
+        ...defaultOptions,
+        ...sdkContext.emitContext.options,
+    };
+    const options = sdkContext.emitContext.options;
     options["models-mode"] = getModelsMode(sdkContext);
     if (options["generate-packaging-files"]) {
         options["package-mode"] = sdkContext.arm ? "azure-mgmt" : "azure-dataplane";
     }
     if (!options["package-name"]) {
-        options["package-name"] = yamlMap["namespace"].replace(/\./g, "-");
+        options["package-name"] = getClientNamespaceStringHelper(sdkContext).replace(/\./g, "-");
     }
+    if (options.flavor !== "azure") {
+        // if they pass in a flavor other than azure, we want to ignore the value
+        options.flavor = undefined;
+    }
+    if (options.flavor) {
+        // if a user passed in value "azure" for options.flavor
+        if (options.unbranded !== undefined) {
+            // if both are set, we ignore unbranded and say we are generating azure code
+            options.unbranded = false;
+        }
+    } else {
+        if (options.unbranded === false) {
+            // this means that a user explicitly passed in unbranded false
+            // we want to get rid of the unbranded flag when passing to the generator
+            // so we just set options.flavor to none, which will do the same thing
+            options.flavor = "azure";
+        }
+        if (sdkContext.emitContext.emitterOutputDir.includes("azure/")) {
```

Comment:
I believe we really see `package-dir` in the `emitterOutputDir` that `tsp` computes for us. However, this means that we're not checking if the `package-dir` starts with `azure`, we just check that the output dir that tsp gives us includes an `azure` folder

Reply:
I trust you on this one